testsuite/forge/src/backend/k8s/node.rs

```diff
@@ -59,9 +63,47 @@ impl K8sNode {
         RestClient::new(self.rest_api_endpoint())
     }
 
-    fn sts_name(&self) -> &str {
+    pub fn sts_name(&self) -> &str {
         &self.sts_name
     }
+
+    fn namespace(&self) -> &str {
+        &self.namespace
+    }
+
+    pub fn spawn_port_forward(&self) -> Result<()> {
+        // spawn a port-forward child process
+        let cmd = Command::new(KUBECTL_BIN)
+            .args([
+                "port-forward",
+                "-n",
+                self.namespace(),
+                &format!("svc/{}", self.dns()),
+                &format!("{}:{}", self.rest_api_port(), REST_API_PORT),
+            ])
+            .stdout(Stdio::null())
+            // .stderr(Stdio::null())
+            .spawn();
+        match cmd {
+            Ok(mut child) => {
+                // sleep a bit and check if port-forward failed for some reason
+                let timeout = Duration::from_secs(1);
+                thread::sleep(timeout);
+                match child.try_wait() {
+                    Ok(Some(status)) => {
+                        info!("Port-forward may have started already: exit {}", status);
+                        Ok(())
+                    }
+                    Ok(None) => {
+                        info!("Port-forward started for {:?}", self);
+                        Ok(())
+                    }
+                    Err(e) => Err(anyhow!("did not work {}", e)),
```

Comment:
Would be good to print the exact equivalent kubectl command that did not work.
E.g. `kubectl port-forward -n forge-5 svc/validator-1 5000:5000` -> Easier to debug failures.

Reply:
sgtm! edited